### DISCUSSION TITLE
Add text-based expression language to QueryBuilder filter

### DISCUSSION THREAD

**Discussion by cweagans** [filamentphp/filament#11731] [Ideas]
The QueryBuilder filter is _excellent_. It's definitely the kind of functionality that I needed for my application. I think there might be a different way of approaching the UI though: what if it was one big text field that accepted a search expression. I was thinking something along the lines of the [Symfony Expression Language](https://github.com/symfony/expression-language). So instead of clicking together a big set of filters (and having to share the resulting link with other people), I could write a search expression like:

```
(email like "%@example.com" or tags contains ["vip"]) and subscribed)
```

to get active subscribers who have an email ending in example.com _or_ who have the `vip` tag.

I could share that expression with somebody pretty easily. I could also potentially build a "Saved searches" system that would just be a way to have a label associated with that expression, then all I'd have to do is fill the search field with that expression.

I think it would also be pretty neat if the QueryBuilder assembled these expressions under the hood. That way, I could have the UI for clicking things together, but the result is a simple text expression.

**Comment by buddhaCode** [comment]
It's possible to persist some of the filters in the URL. Isn't that possible with the query builder? I didn't check that out. But if it got persist in the the URL you could simply share the URL?!

Writing an own query language isn't not that easy.
